tensorflow/tools/pip_package/setup.py

```diff
@@ -295,13 +295,12 @@ def find_files(pattern, root):
         'Intended Audience :: Education',
         'Intended Audience :: Science/Research',
         'License :: OSI Approved :: Apache Software License',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
-        'Programming Language :: Python :: 3.4',
         'Programming Language :: Python :: 3.5',
         'Programming Language :: Python :: 3.6',
         'Programming Language :: Python :: 3.7',
+        'Programming Language :: Python :: 3.8',
+        'Programming Language :: Python :: 3 :: Only',
```

Comment:
What is the difference between `'Programming Language :: Python :: 3'` and `'Programming Language :: Python :: 3 :: Only'`? Do we need both?

Reply:
- `Programming Language :: Python :: 3` means that a project does support Python 3.
- ` Programming Language :: Python :: 3 :: Only` means that a project only supports Python 3, essentially meaning that Python 2 is not supported (any more).

> Do we need both?

If you'd like to have only one of them, go with `Programming Language :: Python :: 3`.

For a similar discussion see https://github.com/numpy/numpy/pull/14372#discussion_r318005631.

https://packaging.python.org/guides/distributing-packages-using-setuptools/#classifiers says:

> Provide a list of classifiers that categorize your project. For a full listing, see https://pypi.org/classifiers/.
> 
> Although the list of classifiers is often used to declare what Python versions a project supports, this information is only used for searching & browsing projects on PyPI, not for installing projects. To actually restrict what Python versions a project can be installed on, use the python_requires argument.

Reply:
Sounds good. Let's have both then